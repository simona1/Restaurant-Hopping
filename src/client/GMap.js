/*global google*/
import React from 'react';
import {Map, Marker, Polyline} from 'google-maps-react';
import {connect} from 'react-redux';
import mapLoaded from './actions/mapLoaded';

const ZOOM = 14;

class GMap extends React.Component {
  componentDidUpdate(prevProps) {
    if (
      this.props.coordinates.latitude === prevProps.coordinates.latitude &&
      this.props.coordinates.longitude === prevProps.coordinates.longitude
    ) {
      // The coordinates didn't change, nothing to do.
      return;
    }
    this._map && this._map.setZoom(ZOOM);
  }

  render() {
    const {coordinates, places} = this.props;
    const {latitude: lat, longitude: lng} = coordinates;

    const path = places.map(place => ({
      lat: place.latitude,
      lng: place.longitude,
    }));

    return (
      <Map
        google={google}
        onReady={this.props.onMapReady}
        zoom={ZOOM}
        center={{lat, lng}}
        initialCenter={{lat, lng}}>
        {places.map((place, index) => (
          <Marker
            key={place.id}
            label={String.fromCharCode('A'.charCodeAt(0) + index % 26)}
            name={place.name}
            title={place.name}
            position={{lat: place.latitude, lng: place.longitude}}
          />
        ))}
        {places.length > 0 && (
          <Polyline
            key={
              /*
                 There is a bug in the implementation of polyline
                 with responding to prop updates, this is a hack
                 that forces unmounting/remounting.
               */
              JSON.stringify(path)
            }
            path={path}
            strokeColor="blue"
            strokeOpacity={0.8}
            strokeWeight={3}
          />
        )}
      </Map>
    );
  }
}

function mapStateToProps(state) {
  return {
    coordinates: state.coordinates,
    places: state.places,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onMapReady: (mapProps, map) => dispatch(mapLoaded(map)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(GMap);
