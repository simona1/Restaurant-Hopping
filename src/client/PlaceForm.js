import PlacesAutocomplete, {
  geocodeByAddress,
  getLatLng,
} from 'react-places-autocomplete';

import React from 'react';
import {connect} from 'react-redux';

import GOOGLE_API_KEY from '../../api_key';

import updateLocationQuery from './actions/updateLocationQuery';

class PlaceForm extends React.Component<Props> {
  render() {
    return (
      <form className="input-field">
        <i className="material-icons prefix">search</i>
        <input
          type="text"
          id="place"
          placeholder="search for establishments nearby"
          value={this.props.locationQuery}
          onChange={this._handleLocationQueryChange}
        />
      </form>
    );
  }

  _handleLocationQueryChange = event => {
    this.props.onLocationQueryChange(event.target.value);
  };
}

function mapStateToProps(state) {
  return {
    locationQuery: state.locationQuery,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onLocationQueryChange: locationQuery =>
      dispatch(updateLocationQuery(locationQuery)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(PlaceForm);
