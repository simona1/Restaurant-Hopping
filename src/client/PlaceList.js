import React from 'react';
import {connect} from 'react-redux';

type Props = {};

class PlacesList extends React.Component<Props> {
  render() {
    // TODO: show something interesting when there are no places yet

    return (
      <ul>
        {this.props.places.map(place => <li key={place.id}>{place.name}</li>)}
      </ul>
    );
  }
}

function mapStateToProps(state) {
  const {places} = state;
  return {places};
}

export default connect(mapStateToProps)(PlacesList);

