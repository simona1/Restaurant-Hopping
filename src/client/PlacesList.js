import React from 'react';
import {connect} from 'react-redux';

type Props = {};

export class PlacesList extends React.Component<Props> {
  render() {
    // TODO: show something interesting when there are no places yet
    return (
      <ol type="A">
        {this.props.places.map(place => (
          <li key={place.id}>
            <p>{place.name}</p>
            {this._renderRating(place.rating, place.id)}
          </li>
        ))}
      </ol>
    );
  }

  _renderRating(rating, id) {
    const stars = [];
    for (let i = 0; i < rating; ++i) {
      stars.push(<i className="material-icons md-icon dp18">grade</i>);
    }
    return (
      <span
        className="item-rating"
        key={id}
        style={{width: Math.floor(24 * rating) + 'px'}}>
        {stars}
      </span>
    );
  }
}

function mapStateToProps(state) {
  const {places} = state;
  return {places};
}

export default connect(mapStateToProps)(PlacesList);
