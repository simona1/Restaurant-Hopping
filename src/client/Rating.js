import React from 'react';

export default class Rating extends React.Component<Props> {
  render() {
    const {rating, id} = this.props;
    const stars = [];
    for (let i = 0; i < rating; ++i) {
      stars.push(
        <i key={`${id}-${i}`} className="material-icons md-icon dp18">
          grade
        </i>,
      );
    }
    return (
      <span
        className="item-rating"
        style={{width: Math.floor(24 * rating) + 'px'}}>
        {stars}
      </span>
    );
  }
}
