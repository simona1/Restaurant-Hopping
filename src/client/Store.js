import {createStore} from 'redux';
import rootReducer from './reducers/root';

const Store = createStore(
  rootReducer,
  {},
  window.devToolsExtension && window.devToolsExtension(),
);

window.Store = Store;

export default Store;
