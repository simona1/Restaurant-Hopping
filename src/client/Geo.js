// @flow

import GMap from './GMap';
import PlaceForm from './PlaceForm';
import PlacesList from './PlacesList';
import React from 'react';
import {Provider} from 'react-redux';

import Store from './Store';

type Props = {};

class Geo extends React.Component<Props> {
  render() {
    return (
      <Provider store={Store}>
        <div>
          <h1>Restaurant Hopping</h1>
          <PlaceForm />
          <div style={{display: 'flex'}}>
            <div
              style={{flexBasis: '80%', position: 'relative', height: '400px'}}>
              <GMap />
            </div>
            <div style={{flexBasis: '20%'}}>
              <PlacesList />
            </div>
          </div>
        </div>
      </Provider>
    );
  }
}

export default Geo;
