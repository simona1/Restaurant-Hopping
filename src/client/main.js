// @flow

import Geo from './Geo';
import React from 'react';
import ReactDOM from 'react-dom';
import nullthrows from 'nullthrows';

type Props = {};

window.onload = function() {
  const appContainer = nullthrows(document.getElementById('app'));
  ReactDOM.render(<Geo />, appContainer);
};
