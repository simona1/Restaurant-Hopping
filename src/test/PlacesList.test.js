import {PlacesList} from '../client/PlacesList';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const testPlaces = [
  {
    id: 'ChIJZ5XfV10GhYARDh2gyITiZGw',
    name: 'Cadet Wine & Beer Bar',
    rating: 4.5,
  },
];

describe('PlacesList', () => {
  it('should be defined', () => {
    expect(PlacesList).toBeDefined();
  });
  it('should render a <PlaceList />', () => {
    const component = shallow(<PlacesList places={testPlaces} />);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should contain a ordered list element', () => {
    const component = shallow(<PlacesList places={testPlaces} />);
    expect(component.contains('ol'));
  });
  it('should contain list item elements', () => {
    const component = shallow(<PlacesList places={testPlaces} />);
    expect(component.contains('li'));
  });
  it('should contain icon elements', () => {
    const component = shallow(<PlacesList places={testPlaces} />);
    expect(component.contains('i'));
  });
  it('should contain paragraph elements', () => {
    const component = shallow(<PlacesList places={testPlaces} />);
    expect(component.contains('p'));
  });
});
