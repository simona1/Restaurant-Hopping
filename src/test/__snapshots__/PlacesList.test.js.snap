// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesList should render a <PlaceList /> 1`] = `
<ol
  type="A"
>
  <li
    key="ChIJZ5XfV10GhYARDh2gyITiZGw"
  >
    <p>
      Cadet Wine & Beer Bar
    </p>
    <span
      className="item-rating"
      style={
        Object {
          "width": "108px",
        }
      }
    >
      <i
        className="material-icons md-icon dp18"
        key="0"
      >
        grade
      </i>
      <i
        className="material-icons md-icon dp18"
        key="1"
      >
        grade
      </i>
      <i
        className="material-icons md-icon dp18"
        key="2"
      >
        grade
      </i>
      <i
        className="material-icons md-icon dp18"
        key="3"
      >
        grade
      </i>
      <i
        className="material-icons md-icon dp18"
        key="4"
      >
        grade
      </i>
    </span>
  </li>
</ol>
`;
